controllers/vmware/vspheremachinetemplate_controller_test.go

```diff
@@ -0,0 +1,185 @@
+		wantStatus             *vmwarev1.VSphereMachineTemplateStatus
+	}{
+		{
+			name:                   "object does not exist",
+			vSphereMachineTemplate: nil,
+			objects:                []client.Object{},
+			wantErr:                false,
+			wantStatus:             nil,
+		},
+		{
+			name:                   "VirtualMachineClass does not exist",
+			vSphereMachineTemplate: vSphereMachineTemplate(namespace.Name, "no-class", "not-existing-class", nil),
+			objects:                []client.Object{},
+			wantErr:                true,
+			wantStatus:             nil,
+		},
+		{
+			name:                   "VirtualMachineClass does exist but has no data",
+			vSphereMachineTemplate: vSphereMachineTemplate(namespace.Name, "with-class", "vm-class", nil),
+			objects: []client.Object{
+				virtualMachineClass(namespace.Name, "vm-class", nil),
+			},
+			wantErr:    false,
+			wantStatus: &vmwarev1.VSphereMachineTemplateStatus{},
+		},
+		{
+			name:                   "VirtualMachineClass does exist and has cpu and memory set",
+			vSphereMachineTemplate: vSphereMachineTemplate(namespace.Name, "with-class", "vm-class", nil),
+			objects: []client.Object{
+				virtualMachineClass(namespace.Name, "vm-class", &vmoprv1.VirtualMachineClassHardware{Cpus: 1, Memory: *resource.NewQuantity(1024, resource.DecimalSI)}),
+			},
+			wantErr: false,
+			wantStatus: &vmwarev1.VSphereMachineTemplateStatus{
+				Capacity: corev1.ResourceList{
+					corev1.ResourceCPU:    quantity(1),
+					corev1.ResourceMemory: quantity(1024),
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			fakeClientBuilder := fake.NewClientBuilder().
+				WithScheme(scheme).
+				WithObjects(append([]client.Object{namespace}, tt.objects...)...)
+
+			vSphereMachineTemplateName := "not-exists"
+			if tt.vSphereMachineTemplate != nil {
+				vSphereMachineTemplateName = tt.vSphereMachineTemplate.GetName()
+				fakeClientBuilder = fakeClientBuilder.
+					WithObjects(tt.vSphereMachineTemplate).
+					WithStatusSubresource(tt.vSphereMachineTemplate)
+			}
+
+			req := reconcile.Request{
+				NamespacedName: types.NamespacedName{
+					Name:      vSphereMachineTemplateName,
+					Namespace: namespace.Name,
+				},
+			}
+
+			r := &vSphereMachineTemplateReconciler{
+				Client: fakeClientBuilder.Build(),
+			}
+
+			vSphereMachineTemplateBefore := &vmwarev1.VSphereMachineTemplate{}
+			if tt.wantStatus != nil {
+				g.Expect(r.Client.Get(ctx, req.NamespacedName, vSphereMachineTemplateBefore)).To(Succeed())
+				_ = vSphereMachineTemplateBefore
+			}
+
+			_, err := r.Reconcile(ctx, req)
+			if (err != nil) != tt.wantErr {
+				g.Expect(err).ToNot(HaveOccurred())
+			}
+
+			if tt.wantStatus != nil {
+				vSphereMachineTemplate := &vmwarev1.VSphereMachineTemplate{}
+				g.Expect(r.Client.Get(ctx, req.NamespacedName, vSphereMachineTemplate)).To(Succeed())
+				g.Expect(vSphereMachineTemplate.Status).To(BeEquivalentTo(*tt.wantStatus))
```

Comment:
```suggestion
				g.Expect(vSphereMachineTemplate.Status).To(BeComparableTo(*tt.wantStatus))
```
just easier to read the diff